src/java/cheshire/prettyprint/CustomPrettyPrinter.java

```diff
@@ -0,0 +1,85 @@
+package cheshire.prettyprint;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+
+import java.io.IOException;
+
+class DynamicIndenter extends DefaultIndenter {
+    public static String repeat(String whitespace, int times) {
+        String indentWith = "";
+        while(--times >= 0) {
+            indentWith += whitespace;
+        }
+        return indentWith;
+    }
+
+    public DynamicIndenter(int indentation) {
```

Comment:
For this, I meant something like:

```java
public DynamicIndenter(String indentString) {
    super(indentString, "\n");
}
```

And then from the clojure side you can do the `repeat` if desired (or pass `"\t"`)